corbel/src/main/scala/graffiti/corbel/auth/AuthorizationDirectives.scala

```diff
@@ -0,0 +1,105 @@
+package graffiti.corbel.auth
+
+import com.google.gson.JsonObject
+import io.corbel.lib.token.TokenInfo
+import io.corbel.lib.token.parser.TokenParser
+import io.corbel.lib.token.reader.TokenReader
+import shapeless.HNil
+import spray.http.HttpMethods
+import spray.routing.directives.BasicDirectives._
+import spray.routing.directives.HeaderDirectives._
+import spray.routing.directives.RouteDirectives._
+import spray.routing.{AuthorizationFailedRejection, Directive0}
+
+import scala.collection.JavaConversions._
+
+/**
+ * @author Alberto J. Rubio
+ */
+trait AuthorizationDirectives {
+
+  val TOKEN_PREFIX = "Bearer"
+
+  def extractToken(value: String): Option[String] = {
+    if (value.startsWith(TOKEN_PREFIX)) {
+      Some(value.substring(TOKEN_PREFIX.length))
+    }
+    else None
+  }
+
+  def extractAuthorizationInfo(parser: TokenParser, token: String): Option[AuthorizationInfo] = {
+    try
+      val tokenReader: TokenReader = parser.parseAndVerify(token)
+      //TODO: Work in progress, we need token access rules
+      //val accessRules: Set[JsonObject] = authorizationRulesService.getAuthorizationRules(token, audience)
+      //if (accessRules != null && accessRules.nonEmpty) {
+      Some(new AuthorizationInfo(tokenReader, Set.empty))
+      //else None
+    catch {
+      case _: Exception => None
+    }
+  }
+
+  def matchesMethod(method: String, rule: JsonObject): Boolean = {
+    rule.get("methods") match {
+      case methods: Any => methods.getAsJsonArray.foreach(json => {
+        if (method.equals(json.getAsString)) {
+          return true
+        }
+      })
+    }
+    false
+  }
+
+  def matchesMediaTypes(accepts: Array[String], rule: JsonObject): Boolean = {
+    rule.get("mediaTypes") match {
+      case mediaTypes: Any => mediaTypes.getAsJsonArray.foreach(json => {
+        if (accepts.contains(json.getAsString)) {
+          return true
+        }
+      })
+    }
+    false
+  }
+
+  def matchesUriPath(uri: String, rule: JsonObject): Boolean = {
+    rule.get("uri") match {
+      case json: Any => uri.matches(json.getAsString)
+      case _ => false
+    }
+  }
+
+  def matchesTokenType(token: TokenInfo, rule: JsonObject): Boolean = {
+    rule.get("tokenType") match {
+      case tokenType: Any => tokenType.getAsString match {
+        case "user" => token.getUserId != null
+        case _ => false
+      }
+      case _ => true
+    }
+  }
+
+  def authorize(parser: TokenParser): Directive0 = {
```

Comment:
The TokenParser dependency can be implicit so that you don't have to to pass it explicitly every time you use the directive